pkg/nsx/services/inventory/builder.go

```diff
@@ -274,16 +275,29 @@ func (s *InventoryService) BuildNamespace(namespace *corev1.Namespace) (retry bo
 		preContainerProject = *preContainerProject.(*containerinventory.ContainerProject)
 	}
 
-	// Extract network errors from annotations
+	// Extract network errors from namespace conditions
 	var networkErrors []common.NetworkError
 	networkStatus := NetworkStatusHealthy
 
-	// Check for VPC error annotation
-	if errorMsg, exists := namespace.Annotations[ctrcommon.AnnotationNamespaceVPCError]; exists && errorMsg != "" {
-		networkErrors = append(networkErrors, common.NetworkError{
-			ErrorMessage: errorMsg,
-		})
-		networkStatus = NetworkStatusUnhealthy
+	// Check for NamespaceNetworkReady condition with status False
+	for _, condition := range namespace.Status.Conditions {
+		if condition.Type == networkinfo.NamespaceNetworkReady && condition.Status == corev1.ConditionFalse {
+			// Create a network error with the condition message
+			networkError := common.NetworkError{
+				ErrorMessage: condition.Message,
+			}
+
+			// Add a reason to the error message if it's one of the known reasons
+			if condition.Reason == networkinfo.NSReasonVPCNetConfigNotReady ||
+				condition.Reason == networkinfo.NSReasonVPCNotReady ||
+				condition.Reason == networkinfo.NSReasonVPCSnatNotReady {
+				networkError.ErrorMessage = fmt.Sprintf("%s: %s", condition.Reason, condition.Message)
```

Comment:
Why not always display `Reason: Message` no matter if it is a known reason? Then we no not need to update the code here when we introduce new reason for vpc

Reply:
It sounds no problem if the reason is not only related to VPC, fixed.